qiskit_ibm_runtime/ibm_backend.py

```diff
@@ -74,13 +158,73 @@ def __init__(
-
+        super().__init__(
+            name=configuration.backend_name,
+            online_date=configuration.online_date,
+            backend_version=configuration.backend_version,
+        )
         self._api_client = api_client
-
-        # Attributes used by caching functions.
+        self._configuration = configuration
         self._properties = None
+        self._qubit_properties = None
         self._defaults = None
+        self._target = None
+        self._max_circuits = configuration.max_experiments
+        if not self._configuration.simulator:
+            self.options.set_validator("noise_model", type(None))
+            self.options.set_validator("seed_simulator", type(None))
+
+    def __getattr__(self, name: str) -> Any:
+        """Gets attribute from self or configuration
+
+        This magic method executes when user accesses an attribute that
+        does not yet exist on IBMBackend class.
+        """
+        # Lazy load properties and pulse defaults and construct the target object.
+        self._get_properties()
+        self._get_defaults()
+        self._convert_to_target()
+        # Check if the attribute now is available on IBMBackend class due to above steps
+        try:
+            return super().__getattribute__(name)
+        except AttributeError:
+            pass
+        # If attribute is still not available on IBMBackend class,
+        # fallback to check if the attribute is available in configuration
+        try:
+            return self._configuration.__getattribute__(name)
+        except AttributeError:
+            raise AttributeError(
+                "'{}' object has no attribute '{}'".format(
+                    self.__class__.__name__, name
+                )
+            )
+
+    def _get_properties(self) -> None:
+        """Gets backend properties and decodes it"""
+        if not self._properties:
+            api_properties = self._api_client.backend_properties(self.name)
+            if api_properties:
+                backend_properties = properties_from_server_data(api_properties)
+                self._properties = backend_properties
+
+    def _get_defaults(self) -> None:
+        """Gets defaults if pulse backend and decodes it"""
+        if not self._defaults and isinstance(
+            self._configuration, PulseBackendConfiguration
+        ):
+            api_defaults = self._api_client.backend_pulse_defaults(self.name)
+            if api_defaults:
+                self._defaults = defaults_from_server_data(api_defaults)
+
+    def _convert_to_target(self) -> None:
+        """Converts backend configuration, properties and defaults to Target object"""
+        if not self._target:
+            self._target = convert_to_target(
+                configuration=self._configuration.to_dict(),
+                properties=self._properties.to_dict() if self._properties else None,
+                defaults=self._defaults.to_dict() if self._defaults else None,
```

Comment:
Why is `to_dict()` necessary here? It's adding extra overhead to the construction. Basically you're going from JSON string response body -> python dict -> `qiskit.providers.models` object -> python dict -> Target and you end up traversing the de-serialized response body 3 times (once for the model object creation, once for the `to_dict()` and once to create the target).

I wrote the converter code assuming it'd be done with the deserialized json payload bodies directly since that'd be the most efficient. But, if that's not feasible here we should modify `convert_to_target` to work with the models objects directly (via attribute access) instead of the output of `to_dict`. This way you at least eliminate one traversal and intermediate object.